spaceapi_server/src/lib.rs

```diff
@@ -140,23 +138,16 @@ fn status_endpoint(_: Request, mut res: Response<Fresh>) {
 
     // Get response body
     let body = build_response_json(people_present, raspi_temperature);
-    let body_bytes = body.as_bytes();
 
-    // Set headers
-    // A new scope is used here because of the mutable borrow.
-    {
-        let mut headers = res.headers_mut();
-        headers.set(header::ContentLength(body_bytes.len() as u64));
-        headers.set(header::ContentType("application/json; charset=utf-8".parse().unwrap()));
-        headers.set(header::CacheControl(vec![header::CacheDirective::NoCache]));
-        headers.set(header::AccessControlAllowOrigin::Any);
-    }
+    // Create response
+    let mut response = Response::with((status::Ok, body));
 
-    // Write response body
-    let mut stream = res.start().unwrap();
-    stream.write_all(body_bytes).unwrap();
+    // Set headers
+    response.set_mut(Header(headers::ContentType("application/json; charset=utf-8".parse().unwrap())));
+    response.set_mut(Header(headers::CacheControl(vec![headers::CacheDirective::NoCache])));
+    response.set_mut(Header(headers::AccessControlAllowOrigin::Any));
```

Comment:
Yes, that's a good point.

Once more, it would be great if we can have a Server struct where we can initialize things like these headers from the constructors. The move to Iron should simplify that.